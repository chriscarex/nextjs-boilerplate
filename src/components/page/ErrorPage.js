import PropTypes from 'prop-types';

const errors = [
  {
    code: 400,
    title: 'Something went wrong',
  },
  {
    code: 403,
    title: 'Unauthorised',
  },
  {
    code: 404,
    title: 'Page not found',
    message: 'If you entered a web address please check it was correct.',
  },
  {
    code: 503,
    title: 'Service unavailable',
    message: 'Sorry, but this service is currently unavailable. Please try again later.',
  }
];

const ErrorPage = ({ statusCode }) => {
  let error = errors.find((e) => e.code === statusCode);
  if (!error) error = errors.find((e) => e.code === 400);

  return (
    <>
      <div>
        <h1 id="error-heading">
          {error.title}
        </h1>
        <div>
          {statusCode === 400 || statusCode === 403
            ? (
              <p>
                Sorry, but an error has occurred with the service.
              </p>
            )
            : <p>{error.message}</p>}
        </div>
      </div>
    </>
  );
};

ErrorPage.propTypes = {
  statusCode: PropTypes.number.isRequired,
};

export default ErrorPage;
