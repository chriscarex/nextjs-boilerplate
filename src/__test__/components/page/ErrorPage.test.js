import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import ErrorPage from '../../../components/page/ErrorPage';

const setup = (statusCode) => {
  const props = {
    statusCode
  };

  const component = shallow(<ErrorPage {...props} />);

  return {
    component
  };
};

describe('ErrorPage', () => {
  it('should match snapshot when 400', () => {
    const {
      component
    } = setup(400);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match snapshot when 403', () => {
    const {
      component
    } = setup(403);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match snapshot when 404', () => {
    const {
      component
    } = setup(404);

    expect(toJson(component)).toMatchSnapshot();
  });

  it('should match snapshot when 503', () => {
    const {
      component
    } = setup(503);

    expect(toJson(component)).toMatchSnapshot();
  });
});
