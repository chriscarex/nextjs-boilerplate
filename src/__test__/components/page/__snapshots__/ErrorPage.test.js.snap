// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should match snapshot when 400 1`] = `
<Fragment>
  <div>
    <h1
      id="error-heading"
    >
      Something went wrong
    </h1>
    <div>
      <p>
        Sorry, but an error has occurred with the service.
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`ErrorPage should match snapshot when 403 1`] = `
<Fragment>
  <div>
    <h1
      id="error-heading"
    >
      Unauthorised
    </h1>
    <div>
      <p>
        Sorry, but an error has occurred with the service.
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`ErrorPage should match snapshot when 404 1`] = `
<Fragment>
  <div>
    <h1
      id="error-heading"
    >
      Page not found
    </h1>
    <div>
      <p>
        If you entered a web address please check it was correct.
      </p>
    </div>
  </div>
</Fragment>
`;

exports[`ErrorPage should match snapshot when 503 1`] = `
<Fragment>
  <div>
    <h1
      id="error-heading"
    >
      Service unavailable
    </h1>
    <div>
      <p>
        Sorry, but this service is currently unavailable. Please try again later.
      </p>
    </div>
  </div>
</Fragment>
`;
